tests/network/bindingplugin_macvtap.go

```diff
@@ -107,11 +128,11 @@ var _ = SIGDescribe("VirtualMachineInstance with macvtap network binding plugin"
 		BeforeEach(checks.SkipIfMigrationIsNotPossible)
 
 		BeforeEach(func() {
-			macAddressHW, err := GenerateRandomMac()
-			Expect(err).ToNot(HaveOccurred())
-			macAddress := macAddressHW.String()
-			clientVMI, err = createAlpineVMIRandomNode(macvtapNetworkName, macAddress)
-			Expect(err).NotTo(HaveOccurred(), "must succeed creating a VMI on a random node")
+			clientVMI = startVMI(libvmi.NewAlpineWithTestTooling(
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(
+					libvmi.InterfaceWithMacvtapBindingPlugin("test"), clientMAC)),
```

Comment:
I remember the mac addresses have been intentionally randomly generated, possibly because of flakes.

I think it was here: https://github.com/kubevirt/kubevirt/commit/5936e7c9e1f8366f351ec278e9fda00f757ef82c

Just removing it without assuring it is safe is risky.
If you find it safe, please reason why it is ok now in the commit message.

Reply:
Since we modify Kubevirt CR (to enable FG, register binding plugin) these test can no longer run in parallel, thus its safe to use static MAC addresses.
Commit message has been updated.

Reply:
> - MAC address generation is replaced with const MAC addressse.
  Since macvtap tests modifies the Kubevirt CR (enable FG, register plugin)
  they cannot run in parallel, thus we can simplify MAC addresses sttings

If I understand correctly, you claim that this problem of having two VMIs at the same existing on the node, with identical mac addresses cannot happen now?

How can you assure that if the VMI deletion is async?
Per what I remember, tests are not waiting for VM/VMI removal before proceeding to the next test. Has this changed?

Reply:
> Its not likely to happen as log the tests not running in parallel.

Why it is not likely too happen?

> Moving aside if its right or not, we do not take care of cleanup objects in tests, we assume the test suite cleans up everything for us.

It cleans them up, but I do not think they may assure deletion before starting a new test.

I am unclear about what you expect here. Are you arguing that we should ignore it or that the cleanup assures that VMIs/Pods are removed before new tests start?

Reply:
I am also unclear what you claim and where this coming from,  you claim we cannot relay on the global test suite cleanup?

Reply:
Tagging you guys again because an edited message wont fire a notification:
@xpivarc , @enp0s3 , could you please confirm this?

Reply:
I tested the pod deletion as the test suite does, delete with zero grace period, and it cause the pod to dispose immediately.
If it happen to take more the few milliseconds enough time will pass until the next pod will create, due to other stuff the test suite does.
If there is a load on the cluster API server that make a pod whose subject to disposal to hangs, creating new pods and other jobs should also hang.

I dont know why there is no check that all pods has been deleted on the clean up, my guess to shorten the time between tests.

I am not willing to drag this PR any longer on questioning basic assumption on the test suite, I will revert the MAC addressing change to be generated instead of static.

Thanks for the discussion.